clang/test/CIR/CodeGen/complex.c

```diff
@@ -303,8 +302,8 @@ void imag_ptr() {
 //      CHECK-AFTER: }
 
 // LLVM: define dso_local void @imag_ptr()
-// LLVM:   store ptr getelementptr inbounds ({ double, double }, ptr @c, i32 0, i32 1), ptr %{{.+}}, align 8
-// LLVM:   store ptr getelementptr inbounds ({ i32, i32 }, ptr @ci, i32 0, i32 1), ptr %{{.+}}, align 8
+// LLVM:   store ptr getelementptr inbounds nuw (i8, ptr @c, i64 8), ptr %{{.+}}, align 8
```

Comment:
https://godbolt.org/z/Yfb4oEcf5

No constant folding for default path, but the maths works out: 1 ptr -> 8 i8

Reply:
oh, I see, this is because `cir-canonicalize`, looks good.

Reply:
Can you add a comment on top of the ones that are constant fold "this differs from traditional codegen because of constant folding".